Source/GameBaseFramework/Private/GAS/Tasks/GBFAT_WaitActorLocationChanged.cpp

```diff
@@ -0,0 +1,108 @@
+#include "GAS/Tasks/GBFAT_WaitActorLocationChanged.h"
+
+FGBFComparisonAxis::FGBFComparisonAxis() :
+    Value( 0.0f ),
+    ComparisonType( EGBFWaitLocationChangeComparisonType::None )
+{
+}
+
+UGBFAT_WaitActorLocationChanged::UGBFAT_WaitActorLocationChanged() :
+    bAllTrue( false )
+{
+    bTickingTask = true;
+}
+
+UGBFAT_WaitActorLocationChanged * UGBFAT_WaitActorLocationChanged::WaitActorLocationChanged(
+    UGameplayAbility * owning_ability,
+    const FGBFComparisonAxis & x_axis_comparison,
+    const FGBFComparisonAxis & y_axis_comparison,
+    const FGBFComparisonAxis & z_axis_comparison,
+    bool all_true /*= true*/ )
+{
+    auto * my_obj = NewAbilityTask< UGBFAT_WaitActorLocationChanged >( owning_ability );
+    my_obj->AxisComparisonX = x_axis_comparison;
+    my_obj->AxisComparisonY = y_axis_comparison;
+    my_obj->AxisComparisonZ = z_axis_comparison;
+    my_obj->bAllTrue = all_true;
+    return my_obj;
+}
+
+void UGBFAT_WaitActorLocationChanged::Activate()
+{
+    Super::Activate();
+
+    SetWaitingOnAvatar();
+
+    if ( GetAvatarActor() == nullptr )
+    {
+        EndTask();
+    }
+}
+
+void UGBFAT_WaitActorLocationChanged::TickTask( const float delta_time )
+{
+    Super::TickTask( delta_time );
+
+    const auto avatar_location = GetAvatarActor()->GetActorLocation();
+
+    const auto x_comparison = GetComparisonResult( AxisComparisonX, avatar_location.X );
+    const auto y_comparison = GetComparisonResult( AxisComparisonY, avatar_location.Y );
+    const auto z_comparison = GetComparisonResult( AxisComparisonZ, avatar_location.Z );
+
+    if ( ( bAllTrue && ( x_comparison && y_comparison && z_comparison ) ) ||
```

Comment:
|| on next line ?

Reply:
clang format will automatically format it like this.

Reply:
this does not compile on clang anyway:

``
GBFAT_WaitActorLocationChanged.cpp(53,28): error: '&&' within '||' [-Werror,-Wlogical-op-parentheses]
         bAllTrue == false && ( x_comparison || y_comparison || z_comparison ) )
         ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
C:\Jenkins\workspace\summercamp_Develop\Plugins\GameBaseFramework\Source\GameBaseFramework\Private\GAS\Tasks\GBFAT_WaitActorLocationChanged.cpp(53,28): note: place parentheses around the '&&' expression to silence this warning
         bAllTrue == false && ( x_comparison || y_comparison || z_comparison ) )
                           ^
         (                                                                    )
```